ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -240,6 +240,17 @@ StreamInfo remove(ClientInvocationId key) {
     return f;
   }
 
+  private void removeDataStream(ByteBuf buf) {
+    try {
+      RaftClientRequest request = ClientProtoUtils.toRaftClientRequest(
+          RaftClientRequestProto.parseFrom(buf.nioBuffer()));
```

Comment:
This works only if request.getType() == Type.STREAM_HEADER.  The code should be:
```
  private void removeDataStream(DataStreamRequestByteBuf request) {
    try {
      final ClientInvocationId invocationId = ClientInvocationId.valueOf(request.getClientId(), request.getStreamId());
      final StreamInfo s = streams.remove(invocationId);
      if (s != null) {
        s.getDivision().getDataStreamMap().remove(invocationId);
      }
    } catch (IOException ignored) {
      LOG.debug(this + ": Failed to removeDataStream for " + request, ignored);
    }
  }
```